Student-Unity/src/main/java/csed/swe/studentunity/Logic/AllCourseService.java

```diff
@@ -64,22 +68,36 @@ public List<Course> getAllActiveCourses(){
         return courseRepo.findCourseByStatus(true);
     }
 
-    public ResponseEntity<Set<Course>> getRegisteredCourses(String sessionId){
+    public ResponseEntity<List<?>> getRegisteredCourses(String sessionId){
         ActiveUserService activeUserService = ActiveUserService.getInstance();
         String userEmail = activeUserService.checkLogin(UUID.fromString(sessionId))[0];
         User user = userService.getUser(userEmail).orElse(null);
         if (user != null)
-            return new ResponseEntity<>(new HashSet<>(), HttpStatus.OK);
-        return new ResponseEntity<>(new HashSet<>(), HttpStatus.NOT_FOUND);
+            return new ResponseEntity<>(registeredCourseRepository.getRegisteredCourseByUserId(user.getId()), HttpStatus.OK);
+        return new ResponseEntity<>(new ArrayList<>(), HttpStatus.NOT_FOUND);
     }
 
-    public HttpStatusCode registerCourse(String sessionId, String courseCode) {
```

Comment:
**Good work I just have a comment**
I think that it is better to return Boolean or enum from the service layer as it is not the beat practice to do so sending it directly from the API layer is better as If we want in the future make the same bake end work with mobile app or we want to change the rest API it will be easily if we decuple the service from the API layer.

Reply:
@AhmedElemary57 
i think it would be more powerful to make the service return a statusCode describes what happened in the service or we can make the service throw meaningful exceptions. (i just sharing my thoughts)

Reply:
@Ahmedelsa3eed
Good idea I will apply it.

Reply:
@Ahmedelsa3eed  
I do not agree with you to make the servicer layer return a status code directly as I said service layer should not speak HTTP language as Is we want in the future make the same back work for an other UI like mobile app or some thing else we should think of making decupling for from the service layer and the API layer.
 
Do the service layer as if you do not know that this service is working for a web application.
this is some references that I read for that.
  
https://www.reddit.com/r/node/comments/uoicw1/should_i_return_status_code_from_service_layer/  

We can also aske the TA for the beast practice

Reply:
@Ahmedelsa3eed  
Yes meaningful expirations but not HTTP responses